File: Desafio #01 - Numero palindromo/PHP/numeroPalindromoCristhianMoreno97.php
<?php
function is_palindrome($number)
{
    if ($number != strrev($number)) {
        return false;
    }
    return true;
}

echo '121: ' . (is_palindrome(121) ? 'Sí' : 'No') . "\n";
echo '1212: ' . (is_palindrome(1212) ? 'Sí' : 'No') . "\n";
echo '1211: ' . (is_palindrome(1211) ? 'Sí' : 'No') . "\n";
echo '12121: ' . (is_palindrome(12121) ? 'Sí' : 'No') . "\n";
echo '121121: ' . (is_palindrome('121121') ? 'Sí' : 'No') . "\n";
echo '121211212112121121211212112121121211212112121: ' . (is_palindrome('121211212112121121211212112121121211212112121') ? 'Sí' : 'No') . "\n";
echo '121211212112121121211212112121121211212112122: ' . (is_palindrome('121211212112121121211212112121121211212112122') ? 'Sí' : 'No') . "\n";

File: Desafio #01 - Numero palindromo/RUST/numeroPalindromoCristhianMoreno97.rs
fn is_palindrome(number: u128) -> bool{
    let mut reversed = 0;
    let mut number_mut = number;

    while number_mut != 0 {
        reversed = reversed * 10 + number_mut % 10;
        number_mut /= 10
    }
    number == reversed
}

fn main() {
    println!("121: {}", is_palindrome(121));
    println!("1212: {}", is_palindrome(1212));
    println!("1211: {}", is_palindrome(1211));
    println!("12121: {}", is_palindrome(12121));
    println!("1200021: {}", is_palindrome(1200021));
    println!("120002112000211200021: {}", is_palindrome(120002112000211200021));
    println!("1200021120002112000211: {}", is_palindrome(1200021120002112000211));
}
